core/src/main/scala/kafka/log/FileMessageSet.scala

```diff
@@ -126,6 +127,8 @@ class FileMessageSet private[kafka](@volatile var file: File,
                        })
   }
 
+  override def asRecords: FileRecords = new FileRecords(file, channel, start, end, isSlice)
```

Comment:
I had this use the more specific type intentionally so that with an instance of `FileMessageSet`, you can get a direct reference to a `FileRecords` instance without any casting. This is convenient for the time being because the capabilities of the file and memory representations are not exactly one-to-one.